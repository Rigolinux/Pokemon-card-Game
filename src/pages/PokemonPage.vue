<!-- eslint-disable prettier/prettier -->
<template>
  <div v-if="!pokemon">
    <h1>Loading...</h1>
  </div>
  <div v-else>
    <h1>How is that pokemon</h1>
    <pokemon-picture :pokemonId="pokemon.id" :showPokemon="showPokemon" />
    <pokemon-options :pokemons="pokemonArr" @selection="checkAnswer($event)" />
    <template v-if="showAnswer">
      <h2>{{ message }}</h2>

      <button @click="newGame">Nuevo Juego</button>
    </template>

    <h3>Aciertos {{ counter }}</h3>
  </div>
</template>
<!-- eslint-disable prettier/prettier -->
<script>
import PokemonOptions from "@/components/PokemonOptions.vue";
import PokemonPicture from "@/components/PokemonPicture.vue";
import getPokemonOptions from "@/helpers/getPokemonOptions";

export default {
  name: "PokemonPage",
  components: {
    PokemonOptions,
    PokemonPicture,
  },
  data() {
    return {
      pokemon: null,
      pokemonArr: [],
      showPokemon: false,
      showAnswer: false,
      message: "",
      counter: 0,
    };
  },
  methods: {
    async mixPokemonArray() {
      this.pokemonArr = await getPokemonOptions();

      const rndInt = Math.floor(Math.random() * this.pokemonArr.length);
      this.pokemon = this.pokemonArr[rndInt];
    },
    checkAnswer(id) {
      this.showPokemon = true;
      if (id === this.pokemon.id) {
        this.showAnswer = true;
        this.message = `Correcto es un ${this.pokemon.name}`;
        this.counter++;
      } else {
        this.showAnswer = true;
        this.message = `Incorrecto, es un ${this.pokemon.name}`;
      }
    },
    newGame() {
      this.showAnswer = false;
      this.showPokemon = false;
      this.pokemon = null;
      this.mixPokemonArray();
    },
  },
  mounted() {
    this.showPokemon = false;

    this.mixPokemonArray();
  },
};
</script>
<!-- eslint-disable prettier/prettier -->
<style></style>
