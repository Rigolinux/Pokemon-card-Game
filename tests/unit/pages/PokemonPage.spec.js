/* eslint-disable prettier/prettier */
import { shallowMount, mount } from "@vue/test-utils";
import PokemonPage from "@/pages/PokemonPage";
import { pokemonsList } from "../mocks/pokemons";

describe("PokemonPage", () => {
  // eslint-disable-next-line no-unused-vars
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonPage);
  });

  test("Debe de hacer match con el snapshot", () => {});

  test("debe llamar mixPokemonArray al montar", () => {
    const mixPokemonArraySpy = jest.spyOn(
      PokemonPage.methods,
      "mixPokemonArray"
    );
    mount(PokemonPage);
    expect(mixPokemonArraySpy).toHaveBeenCalled();
  });

  test("debe de hacer match cuando exista un pokemon", () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemon: pokemonsList[0],
          pokemonArr: pokemonsList,
          showPokemon: false,
          showAnswer: false,
          message: "",
          counter: 0,
        };
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe de mostrar los componentes PokemonPicture y PokemonOptions", () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemon: pokemonsList[0],
          pokemonArr: pokemonsList,
          showPokemon: true,
          showAnswer: true,
          message: "",
          counter: 0,
        };
      },
    });
    const picture = wrapper.find("pokemon-picture-stub");
    const options = wrapper.find("pokemon-options-stub");

    expect(picture.exists()).toBeTruthy();
    expect(options.exists()).toBeTruthy();

    expect(picture.attributes('pokemonid')).toEqual("1");
    expect(options.attributes('pokemons')).not.toBeUndefined();

  });

  test("Pruebas con check answer ",async()=>{
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemon: pokemonsList[0],
          pokemonArr: pokemonsList,
          showPokemon: true,
          showAnswer: true,
          message: "",
          counter: 0,
        };
      },
    });

    //executes the function
/*     await wrapper.vm.checkAnswer(1);
    expect(wrapper.find('h2').exists()).toBeTruthy();
    expect(wrapper.vm.showPokemon).toBeTruthy();
    expect(wrapper.vm.message).toBe(`Correcto es un ${pokemonsList[0].name}`);
 */
    await wrapper.vm.checkAnswer(6);
    expect(wrapper.vm.message).toBe(`Incorrecto, es un ${pokemonsList[0].name}`);
  })
});
