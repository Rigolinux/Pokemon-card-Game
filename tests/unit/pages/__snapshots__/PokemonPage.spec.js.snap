// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage debe de hacer match cuando exista un pokemon 1`] = `
<div>
  <h1>How is that pokemon</h1>
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
  <h3>Aciertos 0</h3>
</div>
`;
